sdks/java/fn-execution/src/main/java/org/apache/beam/sdk/fn/data/CompletableFutureInboundDataClient.java

```diff
@@ -65,6 +65,8 @@ public void complete() {
 
   @Override
   public void fail(Throwable t) {
-    future.completeExceptionally(t);
+    // Use obtrudeException instead of CompleteExceptionally, forcing any future calls to .get()
```

Comment:
Good catch. I did not realize that standard future behavior required this other function :-/